kubernetes/eventhandler.go

```diff
@@ -205,19 +230,31 @@ func (n *nodePodUpdater) OnUpdate(obj interface{}) {
 		return
 	}
 
-	// n.store.List() returns a snapshot at this point. If a delete is received
-	// from the main watcher, this loop may generate an update event after the
-	// delete is processed, leaving configurations that would never be deleted.
-	// Also this loop can miss updates, what could leave outdated configurations.
-	// Avoid these issues by locking the processing of events from the main watcher.
 	if n.locker != nil {
 		n.locker.Lock()
 		defer n.locker.Unlock()
 	}
-	for _, pod := range n.store.List() {
-		pod, ok := pod.(*Pod)
-		if ok && pod.Spec.NodeName == node.Name {
-			n.handler(pod)
+
+	// Slice includes the old and new version of caching object that changes in the current update event. slice[0] is the old version and slice[1] the new updated one
+	slice := n.nodewatcher.Deltaobjects()
+	cachednodeold, ok := slice[0].(*Node)
```

Comment:
`slice[1]` seems to be not used and instead the old version (`slice[0]`) is compared with the `obj.(*Node)` passed to the `OnUpdate`
Do we actually need the updated version?

Reply:
I don't get the meaning of checking the current namespace name is the same as the cached version. Maybe you can clarify that in a comment. Should we instead check the new name has not changed compared to the cached version?

Reply:
@gsantoro  the one you call "current" is the one that triggers the OnUpdate. So still the "current" is not saved in the cache until we make the checks. The actual checks to make the replacement is the ones that come from isEqualMetadata function. And if we have diffrences in metadata we proceed with the Pod updates

FYI : The check we make `ns.Name == cachednamespaceold.Name` is just for safety in order to make sure that we talk for the same object in cache

@tetianakravchenko indeed only the old version is needed (former slice[0]). I initially thought the wather to rerutn only the old version. Then I changed my mind that probably we will need in the future the deltas if we need to make more checks. I dont think that it is a big diff to return both? But tbh I have not tested the memory in scale here

Reply:
@tetianakravchenko I was biased due to [this](https://pkg.go.dev/k8s.io/client-go/tools/cache) and was thinking that we might use it in the future. But not needed for now for sure.

I chanhed again the code only to return old object.